From: Felix Fietkau <nbd@nbd.name>
Date: Wed, 3 Oct 2012 00:00:00 +0200
Subject: [PATCH] mac80211: allow scans in access point mode (for site survey)

Remove the unreachable AP check after the unconditional break; preserve
the existing access-point site-survey policy.
Signed-off-by: ViKing <viking.yfy@qq.com>
---
--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -3508,27 +3508,6 @@ static int ieee80211_scan(struct wiphy *
 		 */
 		fallthrough;
 	case NL80211_IFTYPE_AP:
-		/*
-		 * If the scan has been forced (and the driver supports
-		 * forcing), don't care about being beaconing already.
-		 * This will create problems to the attached stations (e.g. all
-		 * the frames sent while scanning on other channel will be
-		 * lost)
-		 */
-		for_each_link_data(sdata, link) {
-			/* if the link is not beaconing, ignore it */
-			if (!sdata_dereference(link->u.ap.beacon, sdata))
-				continue;
-
-			chan = link->conf->chanreq.oper.chan;
-			radio_idx = cfg80211_get_radio_idx_by_chan(wiphy, chan);
-
-			if (ieee80211_is_radio_idx_in_scan_req(wiphy, req,
-							       radio_idx) &&
-			    (!(wiphy->features & NL80211_FEATURE_AP_SCAN) ||
-			     !(req->flags & NL80211_SCAN_FLAG_AP)))
-				return -EOPNOTSUPP;
-		}
 		break;
 	case NL80211_IFTYPE_NAN:
 	case NL80211_IFTYPE_PD:
